src/system/documents/item.ts

```diff
@@ -375,12 +375,40 @@ export class CosmereItem<
         // Perform the roll
         const roll = await damageRoll(
             foundry.utils.mergeObject(options, {
-                formula: this.system.damage.formula,
+                formula: `${this.system.damage.formula}+${rollData.mod}`,
```

Comment:
If we're going to include the mod in the formula, shouldn't we just require that its actually included in the formula, instead of adding it like this? That would give more freedom for homebrewers, but it would make extracting the base damage harder.

Reply:
I see where you're coming from, but the flipside of such flexibility is that all the standard items then have to actively declare it. Potentially we could achieve the same level of flexibility by providing a checkbox to add the attribute value and default it to on?
Seems like it's a fair assumption for the system to apply this core mechanic to damaging effects (unless this extra modifier only applies to attacks, and other damaging items don't apply this? In which case, default the checkbox to off).
Certainly if I was playing an early release of a system, I'd not be upset that it was making such an assumption even if it meant I couldn't model a particular item I was homebrewing.

Reply:
We should either auto include the mod (as its currently set up to do), or not include it (so it needs to be explicitly added to the formula), anything in between is less clear imo. I think its fine if all standard items have to declare it, I believe that's what 5e does as well. And anyone homebrewing is going to look at the standard items to figure out how to set up their own stuff.
If we auto include the mod there's just certain things that we won't be able to model. I imagine there might be adversaries that can't be easily set up without it.

Also from just a general clarity point; If something is labelled "formula" it should be the formula. The formula shouldn't then be it plus something else. At least not without a hint telling you.

Reply:
I agree with requiring the mod to me manually entered. It's how it's set up in other d20 systems as well

Reply:
Disclaimer: I apologise if my tone in any of this sounds short or harsh, I've just dealt with the mountain/molehill dilemma of getting my nearly 3 year old to bed and I am low on spoons for double checking my sentences. But I do want to find some time to update this with whichever approach we favour tomorrow, so want the discussion to continue. I mean anything said here to be constructive and honestly just see if I have missed something in my understanding.

If you guys prefer to set it to manual, that's fine. Thinking about it, whichever way we implement it, there will be a set of users who assume it's the other way or don't know and just roll something and see how it works - or look at pre-existing examples as you say - in which case it's an easy fix, especially if they have access to a compendium of such examples. I *guess* the expectation for damage values tend to be less for the base roll automation as it is for the basic skill roll; though it's nice if there's an easier way to have it magically do it with few clicks in my experience. And I still am unsure how well-versed the average VTT user is with roll formulas, especially where there's anything platform specific, without a helper/guide. It was doing macros and modding MapTool over 15 years ago that really made me consider programming as a career, and for about half the time since I've been involved with VTTs but I still find my players (who also GM other groups on VTTs) struggle to write out a fairly basic dice pool 🤷🏻‍♂️
If we do go this route however I question whether we need to bother setting a damage stat at all then? might as well just add all the actor attributes to the context the formula needs and then the user can put `@spe` or `@str` or whatever. Alternatively we keep automatic mod addition, but as I suggested above, make it toggleable from the item details window (i.e. a checkbox for "Ignore Modifier?" and have it default to on unless you're creating a weapon?).

> either auto include the mod (as its currently set up to do), or not include it (so it needs to be explicitly added to the formula), anything in between is less clear imo.

Just for my clarity, I believe that the approach I have here *is* an implementation of auto-including the mod. It's not in the middle, I just chose to apply the mod on the roll here rather than in the apply damage step. I feel that made more logical sense as to how I think of the math when doing the rolls manually at an in-person table; it matches closer to how the skill/attack roll is calculated and also pulls similar logic into a closer location in the code, you don't have to go dig to find out where the mod is being added.
If I've missed some implication that makes this different, or even wrong by the rules, I'm sorry I missed it, please let me know 👍🏻

Reply:
(Side Note: I just realise I forgot to remove the apply mod from the damage application in the chat cards as I was telling myself to leave those bits for Mango's changes, so we're doubling the mod on application at the moment! I'll be sure to resolve that one way or another once this is decided!😅)

Reply:
The point IS to allow users who want to to add @spe or @str or whatever they want to the damage formula. By default, we should support @mod to add whatever attribute is used for the skill test, but we shouldn't be hardcoding it to ONLY allow that. A checkbox is fine, but it means if someone wants to skill test with STR then add INT to the damage (for example), they have to first remove the checkbox then add @int manually to the formula. Essentially, like @stanavdb said, the "formula" field becomes not actually the formula, but _part_ of the formula. To me that's even more confusing to setup than just writing "1d6 + @mod" in the damage. 

I do agree that the mod should be included in the actual damage roll object, and not just added later. As I've mentioned elsewhere, my ideal solution for damage rolls would be to have a nested graze damageroll object that can be called directly, so that when a user pressed apply damage we just go "ok full damage is selected, use the base damage roll" or "ok graze damage is selected, use the graze damage roll" and not have to be doing a bunch of logic checks in apply damage functions.

Reply:
unless I'm mistaken by the item setup, we already do separate the damage attribute (red) from the attack one (yellow), and just set the default for the damage to use what is selected for the activation?
![image](https://github.com/user-attachments/assets/c9f9eaf5-47f6-43f3-a2b6-6af125facdff)

I don't know why we have a skill for the damage entry, but I suppose someone might need to use it for something later on.
My comment earlier was that if we don't want to add it automatically we should drop the select boxes in the damage area and just add instructions that explains how to access the attributes/skill ranks through @-notation in the formula; or we keep as is and just add a button to turn off the automatic addition (with `@mod` also being available as a shorthand for the attack modifier so that if you wanted to change both together it would be one input change later).
----
I was trying to get this changeset to end up with the nested roll as you suggested, how far off from what you are after am I? (assuming that the other discussion threads about how we generate the roll values doesn't change the structure of what's done so far, just the values)

Reply:
The object structure atm seems fine to me and in line with how I expected it to be set up. I was just re-iterating that that should stay the case. 

TBH i forgot that we have attribute drop downs in the damage details. I think the sensible thing would be to remove that and have @mod specifically refer to the attribute from the skill test. Then if someone wants to add custom attributes, they can just @ that specific attribute as normal.

Reply:
The attribute dropdown is relevant with regards to graze damage. Surely we only want the mod to drop out for graze damage and not any other flat modifiers? This way the user can target what they want the `@mod` value to be, which is what gets dropped for graze, but still apply any other flat damage based on the roll data.

Reply:
*noises of consternation*
you're right. Assuming we're decided on implementing graze as just "hit damage total - main attribute value". But if we're also deciding that the user has to manually call out adding the damage mod, do we then insist they call out a graze override when they use a mod other than the attack mod?
Or going even further, just ask them to define the graze formula all the time and not have a default for them to override?